nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-nar-utils/src/test/java/org/apache/nifi/nar/NativeLibFinderTest.java

```diff
@@ -0,0 +1,566 @@
+        assertEquals(expected, actual2);
+        verify(testSubjectHelper).findLibrary("libName", nativeLibDir);
+        verify(testSubjectHelper).createTempCopy("libName", "libLocation");
+        verifyNoMoreInteractions(testSubjectHelper);
+    }
+
+    @Test
+    public void testFindLibraryShouldReturnNullWhenLibDirNotRegistered() throws Exception {
+        // GIVEN
+        nativeLibDirs = new ArrayList<>();
+
+        nativeLibNameToPath = new HashMap<>();
+
+        NativeLibFinder testSubject = createTestSubject();
+        String expected = null;
+
+        // WHEN
+        String actual = testSubject.findLibrary("libName");
+
+        // THEN
+        assertEquals(expected, actual);
+        verifyNoMoreInteractions(testSubjectHelper);
+    }
+
+    @Test
+    public void testFindLibraryShouldReturnNullWhenLibNotFound() throws Exception {
+        // GIVEN
+        File nativeLibDir = mock(File.class);
+
+        nativeLibDirs = Arrays.asList(nativeLibDir);
+
+        nativeLibNameToPath = new HashMap<>();
+
+        when(testSubjectHelper.findLibrary("libName", nativeLibDir)).thenReturn(null);
+
+        NativeLibFinder testSubject = createTestSubject();
+        String expected = null;
+
+        // WHEN
+        String actual = testSubject.findLibrary("libName");
+
+        // THEN
+        assertEquals(expected, actual);
+        verify(testSubjectHelper).findLibrary("libName", nativeLibDir);
+        verifyNoMoreInteractions(testSubjectHelper);
+    }
+
+    @Test
+    public void testToDirAppliedShouldReturnParentForFile() throws Exception {
+        // GIVEN
+        Path filePath = createTempFile("mocked").toAbsolutePath();
+        File expected = filePath.getParent().toFile();
+
+        // WHEN
+        File actual = createTestSubject().toDir().apply(filePath.toFile());
+
+        // THEN
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testToDirAppliedShouldReturnDirUnchanged() throws Exception {
+        // GIVEN
+        Path dirPath = createTempFile("mocked").getParent();
+        File expected = dirPath.toFile();
+
+        // WHEN
+        File actual = createTestSubject().toDir().apply(dirPath.toFile());
+
+        // THEN
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetUsrLibDirsShouldReturnUniqueDirs() throws Exception {
+        Path dir1 = Files.createDirectory(tempDirectory.resolve("dir1"));
+        Path dir2 = Files.createDirectory(tempDirectory.resolve("dir2"));
+        Path dir3 = Files.createDirectory(tempDirectory.resolve("dir3"));
+        Path dir4 = Files.createDirectory(tempDirectory.resolve("dir4"));
+
+        Path file11 = createTempFile(dir1, "usrLib", "file11");
+        Path file12 = createTempFile(dir1, "usrLib", "file12");
+        Path file21 = createTempFile(dir2, "usrLib", "file21");
+        Path file31 = createTempFile(dir3, "usrLib", "file31");
+
+        javaLibraryPath = new HashSet<Path>() {{
```

Comment:
Should avoid subclassing HashSet and instead create the HashSet and then add to it. Or, otherwise, use `Arrays.asList(...).stream()` or `Stream.of(...)`